crypto/dh/dh_ameth.c

```diff
@@ -374,13 +374,19 @@ static int dh_cmp_parameters(const EVP_PKEY *a, const EVP_PKEY *b)
 static int int_dh_bn_cpy(BIGNUM **dst, const BIGNUM *src)
 {
     BIGNUM *a;
-    if (src) {
-        a = BN_dup(src);
-        if (!a)
-            return 0;
-    } else
+
+    /*
+     * If read only copy the pointer so we don't have
+     * to reallocate it.
```

Comment:
Could you punctuate the commentary, it's hard to parse...

Reply:
Done.